modules/dynamic-provisioning-azure-file-considerations.adoc

```diff
@@ -34,7 +34,7 @@ parameters:
   location: eastus
   skuName: Standard_LRS
 reclaimPolicy: Delete
-volumeBindingMode: Immediate
+volumeBindingMode: WaitForFirstConsumer
```

Comment:
Based on [this comment](https://bugzilla.redhat.com/show_bug.cgi?id=1918270#c2) in the BZ.

Reply:
@bobfuru IIUC, for azure file, we don't need to set volumeBindingMode to WaitForFirstConsumer, azure file shares should no the limit as azure disk in this comment. The storageclass I recommended uses the correct volumeBindingMode.

Reply:
@qinpingli - I'm confused. The storage class you recommended above for Azure Disk also sets the `volumeBindingMode` to `WaitForFirstConsumer` (https://github.com/openshift/cluster-storage-operator/blob/1842b2a59771d1425272789c080619cad06a2f3c/assets/storageclasses/azure.yaml#L8). Are you saying that I should revert to using `Immediate` for this field?

Reply:
@bobfuru There is some detailed explanation: https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode.

Although from the simple test, looks like `WaitForFirstConsumer` binding mode is harmless for azure file, but this mode is not introduced for azure file.

Reply:
Indeed, for Azure File, `WaitForFirstConsumer` is not better than `Immediate` . It's not worse either.

For AzureDisk we use `WaitForFirstConsumer`, because Azure Disk volumes can be used only by nodes in the same *zone* as the volume, therefore the volume should be provisioned in a zone that has free nodes.

For AzureFile it does not matter that much, because Azure File volumes can be used only by nodes in the same *zone* as the volume can be accessed from any zone and users don't need to worry about volume placement.

Reply:
Thanks both for the thorough explanation. I've reverted to `Immediate` so that this field isn't updated needlessly.